share/src/main/webapp/components/document-details/document-versions.js

```diff
@@ -330,6 +346,9 @@
        */
       onUploadNewVersionClick: function DocumentVersions_onUploadNewVersionClick()
       {
+         this.recordData = this.options.documentDetails.item;
+         this.recordData.jsNode = new Alfresco.util.Node(this.recordData.node);

```

Comment:
My only comment would be that due to JavaScript's copy-by-reference behaviour, it's best to avoid modifying objects that you didn't create - in this case, the `this.options.documentDetails.item` object, which you're inadvertently updating. It also appears that the jsNode object isn't needed outside of this function. This suggestion also means you can remove the `this.recordData` references. 
```javascript
         var jsNode = new Alfresco.util.Node(this.options.documentDetails.item.node);
```